From 2b2033e561bbf35dcdf849b8014be5b9162b8bd6 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 25 Nov 2014 13:23:21 +0100
Subject: [PATCH] multipath/multipathd.service: add more Conditions

Do not start multipathd on rd.multipath=0 rd_NO_MULTIPATH and if there
is no multipath.conf.
---
 modules.d/90multipath/multipathd.service | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules.d/90multipath/multipathd.service b/modules.d/90multipath/multipathd.service
index 44a7b80..b64b02c 100644
--- a/modules.d/90multipath/multipathd.service
+++ b/modules.d/90multipath/multipathd.service
@@ -3,6 +3,10 @@ Description=Device-Mapper Multipath Device Controller
 Before=iscsi.service iscsid.service lvm2-activation-early.service
 DefaultDependencies=no
 Conflicts=shutdown.target
+ConditionKernelCommandLine=!nompath
+ConditionKernelCommandLine=!rd.multipath=0
+ConditionKernelCommandLine=!rd_NO_MULTIPATH
+ConditionPathExists=/etc/multipath.conf
 
 [Service]
 Type=simple
